File: C/6kyu/ArrayDiff.c
/*
DESCRIPTION:
Your goal in this kata is to implement a difference function, which subtracts one list from another and returns the result.

It should remove all values from list a, which are present in list b keeping their order.

array_diff({1, 2}, 2, {1}, 1, *z) == {2} (z == 1)
If a value is present in b, all of its occurrences must be removed from the other:

array_diff({1, 2, 2, 2, 3}, 5, {2}, 1, *z) == {1, 3} (z == 2)
NOTE: In C, assign return array length to pointer *z
*/

#include <stdlib.h>
#include <stdbool.h>
bool isin (int num,int *arr , size_t sz)
{
  for(int i = 0 ; i < sz ; i++)
    if(num == arr[i])
      return true;
  return false;
}

int *array_diff(const int *arr1, size_t n1, const int *arr2, size_t n2, size_t *z) {
  int *arr = (int *) malloc(sizeof(int) * n1);
  for(int i = 0 ; i < n1 ; i++)
    if(!isin(arr1[i] , arr2 , n2))
      arr[(*z)++] = arr1[i];
  return arr;
}

File: java/6kyu/ArrayDiff.java
/*
DESCRIPTION:
Your goal in this kata is to implement a difference function, which subtracts one list from another and returns the result.

It should remove all values from list a, which are present in list b keeping their order.

Kata.arrayDiff(new int[] {1, 2}, new int[] {1}) => new int[] {2}
If a value is present in b, all of its occurrences must be removed from the other:

Kata.arrayDiff(new int[] {1, 2, 2, 2, 3}, new int[] {2}) => new int[] {1, 3}
*/

public class Kata {

      public static boolean isin (int arr[] , int num)
    {
        for(int i = 0 ; i <arr.length ; i++)
        {
            if(num == arr[i])
                return true;
        }
        return false;
    }

    public static int findsize (int a[] , int b[])
    {
        int size = 0;
        for (int i = 0 ; i < a.length ; i++)
        {
            if(!isin(b,a[i]))
            {
                size++;
            }
        }
        return size;
    }

    public static int[] arrayDiff(int[] a, int[] b) {
        int c[] = new int[findsize(a,b)];
        int index = 0;
        for (int i = 0 ; i < a.length ; i++)
        {
            if(!isin(b,a[i]))
            {
                c[index++] = a[i];
            }
        }
        for (int i = 0 ; i < c.length ; i++)
            System.out.println(c[i]);
        return c;
    }

}
